train.py

```diff
@@ -192,84 +319,144 @@ def inference(engine, batch):
         events |= Events.STARTED
 
     @trainer.on(events)
-    def run_evalutation():
+    def run_evaluation():
         evaluator.run(val_loader)
 
+    trainer.add_event_handler(
+        Events.EPOCH_COMPLETED | Events.COMPLETED | Events.STARTED, run_evaluation
+    )
     # Make sure distributed data samplers split the dataset nicely between the distributed processes
-    if args.distributed:
-        trainer.add_event_handler(Events.EPOCH_STARTED, lambda engine: train_sampler.set_epoch(engine.state.epoch))
-        evaluator.add_event_handler(Events.EPOCH_STARTED, lambda engine: valid_sampler.set_epoch(engine.state.epoch))
-
-    # Linearly decrease the learning rate from lr to zero
-    scheduler = PiecewiseLinear(optimizer, "lr", [(0, args.lr), (args.n_epochs * len(train_loader), 0.0)])
-    trainer.add_event_handler(Events.ITERATION_STARTED, scheduler)
-
-    # Prepare metrics - note how we compute distributed metrics
-    RunningAverage(output_transform=lambda x: x).attach(trainer, "loss")
-    metrics = {"nll": Loss(torch.nn.CrossEntropyLoss(ignore_index=-100), output_transform=lambda x: (x[0][0], x[1][0])),
-               "accuracy": Accuracy(output_transform=lambda x: (x[0][1], x[1][1]))}
-    metrics.update({"average_nll": MetricsLambda(average_distributed_scalar, metrics["nll"], args),
-                    "average_accuracy": MetricsLambda(average_distributed_scalar, metrics["accuracy"], args)})
-    metrics["average_ppl"] = MetricsLambda(math.exp, metrics["average_nll"])
-    for name, metric in metrics.items():
-        metric.attach(evaluator, name)
+    if args.dist_backend is not None:
+        trainer.add_event_handler(
+            Events.EPOCH_STARTED,
+            lambda engine: train_sampler.set_epoch(engine.state.epoch),
+        )
+        evaluator.add_event_handler(
```

Comment:
I think this is useless as there is no shuffling for validation dataloader... But they do that in their code. Let's keep in mind and remove it in another PR.